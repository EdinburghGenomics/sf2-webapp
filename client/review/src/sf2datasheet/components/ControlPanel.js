// @flow
import React from 'react';


// Flow type declarations ----

type ControlPanelProps = {
    uuid: string,
    onFillRow: (up: boolean) => void,
    onFillCell: (up: boolean) => void,
    disabled: boolean,
    showFillUp: boolean
};


// React component ----

const ControlPanel = (props : ControlPanelProps) => {


    const handleFillDownRowClick = e => {
        e.preventDefault();
        props.onFillRow(false);
    };


    const handleFillDownCellClick = e => {
        e.preventDefault();
        props.onFillCell(false);
    };


    const handleFillUpRowClick = e => {
        e.preventDefault();
        props.onFillRow(true);
    };


    const handleFillUpCellClick = e => {
        e.preventDefault();
        props.onFillCell(true);
    };


    return (
        <div>
            <button disabled={props.disabled} data-uuid={props.uuid} onClick={handleFillDownRowClick}>Fill down row</button>
            <button disabled={props.disabled} data-uuid={props.uuid} onClick={handleFillDownCellClick}>Fill down cell</button>
            {props.showFillUp && <button disabled={props.disabled} data-uuid={props.uuid} onClick={handleFillUpRowClick}>Fill up row</button>}
            {props.showFillUp && <button disabled={props.disabled} data-uuid={props.uuid} onClick={handleFillUpCellClick}>Fill up cell</button>}
        </div>
    );

};


//$FlowFixMe
export default ControlPanel;
