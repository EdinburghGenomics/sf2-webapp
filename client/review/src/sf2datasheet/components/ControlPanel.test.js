import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import ControlPanel from "./ControlPanel";


// Basic page logic

it('renders Control Panel correctly', () => {

    const wrapper = mount(<ControlPanel disabled={false} uuid={"1"} onFillRow={up => {}} onFillCell={up => {}} showFillUp={true}/>);

    expect(toJson(wrapper)).toMatchSnapshot();

});
