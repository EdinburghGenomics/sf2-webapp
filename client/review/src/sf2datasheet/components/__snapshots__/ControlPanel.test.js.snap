// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Control Panel correctly 1`] = `
<ControlPanel
  disabled={false}
  onFillCell={[Function]}
  onFillRow={[Function]}
  showFillUp={true}
  uuid="1"
>
  <div>
    <button
      data-uuid="1"
      disabled={false}
      onClick={[Function]}
    >
      Fill down row
    </button>
    <button
      data-uuid="1"
      disabled={false}
      onClick={[Function]}
    >
      Fill down cell
    </button>
    <button
      data-uuid="1"
      disabled={false}
      onClick={[Function]}
    >
      Fill up row
    </button>
    <button
      data-uuid="1"
      disabled={false}
      onClick={[Function]}
    >
      Fill up cell
    </button>
  </div>
</ControlPanel>
`;
